xpk.py

```diff
@@ -4072,6 +4076,183 @@ def install_nccl_on_cluster(args) -> int:
+  valid_master_versions: set[str]
+  valid_node_versions: set[str]
+
+
+def get_gke_server_config(args) -> tuple[int, GkeServerConfig | None]:
+  """Determine the GKE versions supported by gcloud currently.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    Tuple of
+    int: 0 if successful and 1 otherwise.
+    GkeServerConfig: stores valid gke version to use in node pool and cluster.
+  """
+  base_command = (
+      'gcloud container get-server-config'
+      f' --project={args.project} --region={zone_to_region(args.zone)}'
+  )
+  default_rapid_gke_version_cmd = (
+      base_command
+      + " | awk '/RAPID/,/defaultVersion/' | awk 'FNR == 2 {print $2}' "
+  )
+  valid_master_versions_cmd = (
+      base_command
+      + " | awk '/validMasterVersions/,/validNodeVersions/' | awk '/-/,/-/"
+      " {print $2}' "
+  )
+  valid_node_versions_cmd = (
+      base_command
+      + " | awk '/validNodeVersions/,/END/' | awk '/-/,/-/ {print $2}' "
+  )
+  base_command_description = 'Determine server supported GKE versions for'
+
+  task = 'default rapid gke version'
+  return_code, default_rapid_gke_version = run_command_for_value(
+      default_rapid_gke_version_cmd, f'{base_command_description} {task}.', args
+  )
+  if return_code != 0:
+    xpk_print(f'Unable to get server config for {task}.')
+    return return_code, None
+
+  task = ' valid master versions'
+  return_code, valid_master_versions = run_command_for_value(
+      valid_master_versions_cmd, f'{base_command_description} .', args
+  )
+  if return_code != 0:
+    xpk_print(f'Unable to get server config for {task}.')
+    return return_code, None
+
+  task = 'valid node versions'
+  return_code, valid_node_versions = run_command_for_value(
+      valid_node_versions_cmd, f'{base_command_description} .', args
+  )
+  if return_code != 0:
+    xpk_print(f'Unable to get server config for {task}.')
+    return return_code, None
+
+  return 0, GkeServerConfig(
+      default_rapid_gke_version=default_rapid_gke_version,
+      valid_master_versions=set(valid_master_versions.splitlines()),
+      valid_node_versions=set(valid_node_versions.splitlines()),
+  )
+
+
+def get_gke_control_plane_version(
+    args, gke_server_config: GkeServerConfig
+) -> tuple[int, str | None]:
+  """Determine gke control plane version for cluster creation.
+
+  Args:
+    args: user provided arguments for running the command.
+    gke_server_config: holds valid gke versions and recommended default version.
+
+  Returns:
+    Tuple of
+    int: 0 if successful and 1 otherwise.
+    str: gke control plane version to use.
+  """
+
+  # Override with user provide gke version if specified.
+  if args.gke_version is not None:
+    master_gke_version = args.gke_version
+  else:
+    master_gke_version = gke_server_config.default_rapid_gke_version
+
+  is_valid_master_version = (
+      master_gke_version in gke_server_config.valid_master_versions
+  )
+  is_valid_node_version = (
```

Comment:
Why do we need to check the node pool gke version here? Isn't `get_gke_node_pool_version()` doing this?

Reply:
This is needed for cpu node pools created through cluster create command.